docs/sql/listinstances/listinstances.go

```diff
@@ -0,0 +1,30 @@
+package listinstances
+
+import (
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/sqladmin/v1beta4"
+)
+
+func ListInstances(projectId string) ([]*sqladmin.DatabaseInstance, error) {
+	// Create a client that uses application's default credentials.
+	client, err := google.DefaultClient(
+		oauth2.NoContext,
+		"https://www.googleapis.com/auth/sqlservice.admin")
+	if err != nil {
+		return nil, err
+	}
+	// Create the Google Cloud SQL service.
+	service, err := sqladmin.New(client)
+	if err != nil {
+		return nil, err
+	}
+	// Create the request to list instances for the project ID.
+	request := service.Instances.List(projectId)
```

Comment:
Done. And cleaned up a bunch of other stuff (like using the scope constant, variable names)